packages/nimble-components/src/select/index.ts

```diff
@@ -471,20 +473,23 @@ export class Select
         this.clearSelection();
         this.filterOptions();
 
-        if (this.filteredOptions.length > 0) {
-            const enabledOptions = this.filteredOptions.filter(
-                o => !o.disabled
-            );
-            if (enabledOptions.length > 0) {
-                enabledOptions[0]!.selected = true;
-            } else {
-                // only filtered option is disabled
-                this.selectedOptions = [];
-                this.selectedIndex = -1;
-            }
-        } else if (this.committedSelectedOption) {
-            this.committedSelectedOption.selected = true;
+        const enabledOptions = this.filteredOptions.filter(o => !o.disabled);
+        let selectedOptionIndex = this.filter !== ''
```

Comment:
Perhaps `selectedOptionIndex` should be `activeOptionIndex` or `updatedActiveOptionIndex`. Calling it "selected" is a bit misleading since the option isn't actually being selected.